### ISSUE TITLE
Cannot set property 'socket' of null

### ISSUE THREAD

**Issue by tsonevn** [NativeScript/nativescript-cli#3818]
_From @RoyiNamir on August 12, 2018 9:42_

After upgrading our project to 4.2 , we've experiencing socket connections problems to a real device  when `tns run android`

Worth to mention that `tns run android` **does** work with emulators ( both Geny & AVD) , but it doesn't work with real devices ( my Galaxy s8+ and a colleague's Galaxy s8 , non rooted regular device)

```
C:\projects-web-storm\digital\dsapp1>tns info
√ Getting NativeScript components versions information...
√ Component nativescript has 4.2.1 version and is up to date.
√ Component tns-core-modules has 4.2.0 version and is up to date.
√ Component tns-android has 4.2.0 version and is up to date.
× Component tns-ios is not installed.
```

When `tns run android` : 

we get : 

```
Unable to apply changes on device: ce031713a8d0149e0d. Error is: Socket connection timeouted..
Executing after-watch hook from C:\projects-web-storm\digital\dsapp1\hooks\after-watch\nativescript-dev-sass.js
Executing after-watch hook from C:\projects-web-storm\digital\dsapp1\hooks\after-watch\nativescript-dev-typescript.js
Stopping tsc watch
Executing after-watch hook from C:\projects-web-storm\digital\dsapp1\hooks\after-watch\nativescript-dev-webpack.js
TypeError: Cannot set property 'socket' of null
    at Timeout.tryConnect [as _onTimeout] (C:\Users\royin\AppData\Roaming\npm\node_modules\nativescript\lib\services\livesync\android-livesync-tool.js:298:51)
    at ontimeout (timers.js:482:11)
    at tryOnTimeout (timers.js:317:5)
    at Timer.listOnTimeout (timers.js:277:5)
```

[**--> Video showing the problem <--** ](https://imgur.com/a/TQItAHG)

The app IS installed **but not invoked**. ( it used to work) before upgrade.
We have to tap the launch icon in order to see the app running
PackageID is the **same** in all files ( manifest / packagejson)

BTW - if you want a real project that produces it , I can send privately. 

-----------

Windows machine , Android , Antivirus  & [firewall  ](https://i.imgur.com/WzKMUo3.jpg)is disabled , uninstalled App from device.

_Copied from original issue: NativeScript/NativeScript#6180_

**Comment by farfromrefug** [comment]
@Fatme  @tsonevn I am still seeing this with 4.2.2
The first run  works. But then if I run it again it fails.
In my logcat I keep on seeing this:
failed to connect to socket 'localabstract:org.nativescript.apps-livesync': Connection refused

I am getting this almost all the time :s I can't even update my apps.  
Happening on emulator and device

**Comment by RoyiNamir** [comment]
@farfromrefug remove the dash from :org.nativescript.apps-livesync and make a capital letter in org.nativescript.Appslivesync

**Comment by farfromrefug** [comment]
@RoyiNamir and where should I do that. I can't This I not the app name! The app name is simply ```org.nativescript.apps```
I guess the ```-livesync``` is added by the CLI

**Comment by RoyiNamir** [comment]
@farfromrefug can you paste your android manifest file?

**Comment by farfromrefug** [comment]
[manifest](https://github.com/NativeScript/NativeScript/blob/master/apps/app/App_Resources/Android/AndroidManifest.xml) ;)

**Comment by RoyiNamir** [comment]
Are you extending an activity also?

**Comment by farfromrefug** [comment]
no I simply run the ui test app :D working on the framework

**Comment by RoyiNamir** [comment]
@farfromrefug if that's the case then i guess i dont know...

**Comment by nikopolv** [comment]
I did yea. I also tried com.company.android.app, com.company.AndroidApp, Com.Company.AndroidApp
The same results.

> Project successfully built.
Installing...
Successfully installed on device with identifier 'xxxxx'.
Unable to apply changes on device: xxxxx. Error is: Socket connection timeouted..
Executing after-watch hook from C:\...\hooks\after-watch\nativescript-dev-typescript.js
Stopping tsc watch
TypeError: Cannot set property 'socket' of null
    at Timeout.tryConnect [as _onTimeout] (C:\....\npm\node_modules\nativescript\lib\services\livesync\android-livesync-tool.js:298:51)
    at ontimeout (timers.js:498:11)
    at tryOnTimeout (timers.js:323:5)
    at Timer.listOnTimeout (timers.js:290:5)

**Comment by RoyiNamir** [comment]
@nikopolv  the rule (after 4.2.2) is `*.*.[capital]*`

Also did you remove platform and then add ?

**Comment by DimitarTachev** [comment]
Hi @farfromrefug , @RoyiNamir, @nikopolv,

This exception should not be caused by your app ids. We don't have a rule for a capital letter there.  The starting capital letter is required only for the class name of the custom activities and there aren't any changes there. In other words, `org.nativescript.apps` is a perfectly valid app id and the sample is working correctly with NativeScript `4.2.2` on our side.

The real issue looks like the above-mentioned `failed to connect to socket 'localabstract:org.nativescript.apps-livesync': Connection refused` exception.  @farfromrefug @nikopolv, could you run `tns run android --log trace` and update us back with the detailed log from the problematic executions? I also suggest you make a clean copy of the `org.nativescript.apps` app in order to avoid getting some strange errors caused by a parcial app state after the experiments.

@RoyiNamir, thanks for helping with that, however, [the regex change in 4.2.2 ](https://github.com/telerik/mobile-cli-lib/compare/9100f6cc828c6bc2c585c67c2d550425e907b7d8...1383f3e88b873cefac8a0c6738ad139727e1f0ed#diff-4576ea61b5a71205094e7ee648cbf5e1R155)is just allowing the usage of capital letters in the packages of the custom activities (e.g. using `org.nativescript.apps/org.MyCoolApp.MyCoolActivity` instead of [the recommended ](https://en.wikipedia.org/wiki/Java_package#Package_naming_conventions)`org.nativescript.apps/org.mycoolapp.MyCoolActivity`) and I don't belive this change is related to the issues of @farfromrefug and @nikopolv.

**Comment by RoyiNamir** [comment]
@DimitarTachev Thanks for clarifying. However , I've _never_ mentioned app id. I always asked about `activity name ` ;-)

However , it might be that i've confused (miscommunication)  appID vs activity name .

BTW - as you can see in the OP's log  , there is a timeout socket , and my( spcific) problem was activity name named `xxx.myApp.Something` ( problematic A)

**Comment by DimitarTachev** [comment]
@RoyiNamir, the app id clarification was related to the "org.nativescript.apps-livesync" comments above. That's just a socket related string containing the app id concatenated with "-livesync" and it's not causing the exception.

Regarding your specific problem, you should be also able to use `xxx.myApp.Something` with NativeScript 4.2.2 but as I already mentioned, the lowercase package name is the recommended approach :)

We will need more info from @farfromrefug and @nikopolv in order to further investigate their issues as I don't believe thay are related to custom activities with some strange naming. I hope we will get more details for the `failed to connect to socket 'localabstract:org.nativescript.apps-livesync': Connection refused` error from their trace logs.

**Comment by manojdcoder** [comment]
@tsonevn This also happens when the nativescript id in `package.json` is different from one in the manifest / gradle file. I had to keep them different as I wanted different package names for iOS & Android.

**Comment by aaayushsingh** [comment]
Still getting this after upgrading to `4.2.3`, plus a new error as well see #3851

**Comment by jannotabamo** [comment]
Im still getting the error after upgrading to 4.2.4. Any workaround for this issue?

**Comment by NathanWalker** [comment]
I just started getting this error as well today with Android - very strange 🤔

**Comment by tsonevn** [comment]
Hi, @jannotabamo @NathanWalker,
Can you run the project with `--log trace` flag (`tns run <platform_name> --log trace`) and send us the full log with the error? Also, can you provide more info, if this problem happens consistently and whether this issue appears only on a specific device or emulators?
Meanwhile, try to clear up the project by deleting `platforms`, `node_modules` and `hooks` folders and check if you will be able to recreate the problem.

@aaayushsingh, can you upgrade to NativeScript CLI `4.2.4` and check if the problem will be resolved on your side?

**Comment by shiv19** [comment]
@tsonevn @DimitarTachev 
Hope this log file helps. 🤞🏻
[log.txt](https://github.com/NativeScript/nativescript-cli/files/2457732/log.txt)

**Comment by NathanWalker** [comment]
Thanks @shiv19 hopefully @tsonevn that log helps 🤞

**Comment by KristianDD** [comment]
Hi @shiv19 ,
I have doubled the time we wait for the socket connection to be established in the master branch. You can try the fix with the `next` version. Just install CLI like this `npm i -g nativescript@next`.

If that doesn't help please check if the application is started on the device, before the message appears. If the application is not started at all unify the values of
1 `nativescript.id` key in `package.json` 
2 `applicationId` in `app/App_Resources/Android/app.gradle` 
3 `appId` in `app/App_Resources/Android/settings.json`(this one might not exist if you don't use NS Sidekick)
It they are different the CLI can't work properly.

**Comment by shiv19** [comment]
@nwalker
See that 3rd point in previous reply. 
That explains how we have settings.json in that project. 
I used sidekick on that project recently when we changed the icons.

**Comment by shiv19** [comment]
@KristianDD 
Thank you! Looks like that has solved the issue. I don't see that problem anymore.

**Comment by rosen-vladimirov** [comment]
Hey everyone,
We've merged an improvement in the error message shown by CLI in this case that suggests some ideas what might cause the error. The new message is available in CLI's rc version (`npm i -g nativescript@rc`) and should be released with CLI 6.3.3 version.
